src/js_parser.zig

```diff
@@ -23102,677 +23059,355 @@ fn NewParser_(
+                    @as(usize, @intFromBool(have_custom_hooks)),
+            ) catch bun.outOfMemory();
 
-                                .s_export_star, .s_export_from => {
-                                    exports_list[exports_list_i] = stmt.*;
-                                    stmt.loc = exports_list[exports_list_i].loc;
-                                    exports_list_i += 1;
-                                },
-                                else => {
-                                    remaining_final_stmts[0] = stmt.*;
-                                    remaining_final_stmts = remaining_final_stmts[1..];
-                                },
-                            }
-                            stmt.* = Stmt.empty();
-                        }
-                    }
+            args[0] = function_with_hook_calls;
+            args[1] = p.newExpr(E.String{ .data = hash_data }, loc);
 
-                    commonjs_wrapper.data.e_call.args.ptr[0] = p.newExpr(
-                        E.Function{ .func = G.Fn{
-                            .name = null,
-                            .open_parens_loc = logger.Loc.Empty,
-                            .args = require_function_args,
-                            .body = .{ .loc = logger.Loc.Empty, .stmts = final_stmts_list },
-                            .flags = Flags.Function.init(.{ .is_export = true }),
-                        } },
-                        logger.Loc.Empty,
-                    );
-                    var sourcefile_name = p.source.path.pretty;
-                    if (strings.lastIndexOf(sourcefile_name, "node_modules")) |node_modules_i| {
-                        // 1 for the separator
-                        const end = node_modules_i + 1 + "node_modules".len;
-                        // If you were to name your file "node_modules.js" it shouldn't appear as ".js"
-                        if (end < sourcefile_name.len) {
-                            sourcefile_name = sourcefile_name[end..];
-                        }
-                    }
-                    commonjs_wrapper.data.e_call.args.ptr[1] = p.newExpr(E.String{ .data = sourcefile_name }, logger.Loc.Empty);
+            if (have_force_arg) args[2] = p.newExpr(E.Boolean{ .value = p.react_refresh.force_reset }, loc);
 
-                    new_stmts_list[imports_list.len] = p.s(
-                        S.ExportDefault{
-                            .value = .{
-                                .expr = commonjs_wrapper,
-                            },
-                            .default_name = LocRef{ .ref = null, .loc = logger.Loc.Empty },
-                        },
-                        logger.Loc.Empty,
-                    );
-                    parts[parts.len - 1].stmts = new_stmts_list;
-                },
+            if (have_custom_hooks) {
+                // () => [useCustom1, useCustom2]
+                args[3] = p.newExpr(E.Arrow{
```

Comment:
this is out of bounds if `!have_force_arg` and `have_custom_hooks`?

Reply:
not possible because have_force_arg is defined in terms of have_custom_hooks

```zig
const have_custom_hooks = ctx.user_hooks.count() > 0;
const have_force_arg = have_custom_hooks or p.react_refresh.force_reset;
```